src/native/crt.c

```diff
@@ -231,8 +231,13 @@ JNIEnv *aws_jni_get_thread_env(JavaVM *jvm) {
 #else
         jint result = (*jvm)->AttachCurrentThreadAsDaemon(jvm, (void **)&env, NULL);
 #endif
-        (void)result;
-        AWS_FATAL_ASSERT(result == JNI_OK);
+        /* Ran out of memory, don't log in this case */
+        AWS_FATAL_ASSERT(result != JNI_ENOMEM);
+        if (result != JNI_OK) {
+            /* Result in crash, but log the return code first. */
+            AWS_LOGF_ERROR(AWS_LS_COMMON_GENERAL, "AttachCurrentThreadAsDaemon failed, error code is %d", result);
+            AWS_FATAL_ASSERT(result == JNI_OK);
```

Comment:
does our async logger have time to actually spit this out before the FATAL_ASSERT kills the program?

would `fprintf(stderr, ...)` be better?